scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt

```diff
@@ -0,0 +1,128 @@
+import org.ossreviewtoolkit.model.TextLocation
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.ClearlyDefinedStorageConfiguration
+
+class ClearlyDefinedStorageFunTest : StringSpec({
+    val storage = ClearlyDefinedStorage(ClearlyDefinedStorageConfiguration(ClearlyDefinedService.Server.PRODUCTION.url))
+
+    "Scan results for ScanCode 3.0.2 should be read correctly" {
+        val id = Identifier("Maven:com.vdurmont:semver4j:3.1.0")
+        val result = storage.read(id)
+
+        result.shouldBeSuccess {
+            it shouldContain ScanResult(
+                provenance = ArtifactProvenance(
+                    sourceArtifact = RemoteArtifact(
+                        url = "https://search.maven.org/remotecontent" +
+                                "?filepath=com/vdurmont/semver4j/3.1.0/semver4j-3.1.0-sources.jar",
+                        hash = Hash(
+                            value = "0de1248f09dfe8df3b021c84e0642ee222cceb13",
+                            algorithm = HashAlgorithm.SHA1
+                        )
+                    )
+                ),
+                scanner = ScannerDetails(
+                    name = "ScanCode",
+                    version = "3.0.2",
+                    configuration = "input /tmp/cd-2rGiCR --classify true --copyright true --email true " +
+                            "--generated true --info true --is-license-text true --json-pp /tmp/cd-0EjTZ7 " +
+                            "--license true --license-clarity-score true --license-diag true --license-text true " +
+                            "--package true --processes 2 --strip-root true --summary true --summary-key-files true " +
+                            "--timeout 1000.0 --url true"
+                ),
+                summary = ScanSummary(
+                    startTime = Instant.parse("2020-02-14T00:36:14.000335513Z"),
+                    endTime = Instant.parse("2020-02-14T00:36:37.000492119Z"),
+                    packageVerificationCode = "",
+                    licenseFindings = sortedSetOf(
+                        LicenseFinding(
+                            license = "MIT",
+                            location = TextLocation(
+                                path = "META-INF/maven/com.vdurmont/semver4j/pom.xml",
+                                startLine = 30,
+                                endLine = 31
+                            ),
+                            score = 60.87f
+                        )
+                    ),
+                    copyrightFindings = sortedSetOf(),
+                    issues = emptyList()
+                )
+            )
+        }
+    }
+
+    "Scan results for ScanCode 30.1.0 should be read correctly" {
+        val id = Identifier("Maven:com.sksamuel.hoplite:hoplite-core:2.1.3")
+        val result = storage.read(id)
+
+        result.shouldBeSuccess {
+            it shouldContain ScanResult(
```

Comment:
`shouldContainExactly`?

Reply:
My idea here was that newly added results on CD should not make the text break. Same below.